### PR TITLE
metrics: add doublesign counter

### PR DESCRIPTION
**Pull request by emailtovamos**
### Description

double sign counter metric

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: consensus/parlia/parlia.go [hunk 1]
```diff
@@ -2011,3 +2031,17 @@ func applyMessage(
 	}
 	return msg.Gas() - returnGas, err
 }
+
+// proposalKey build a key which is a combination of the block number and the proposer address.
+func proposalKey(header types.Header) string {
+
+	slotKey := uintToString(header.Number.Uint64())
+	proposerIndexKey := uintToString(header.Coinbase.Big().Uint64())
```
Comment (zzzckck) [inline consensus/parlia/parlia.go:2039]: `header.Coinbase.String()`?

File: consensus/parlia/parlia.go [hunk 2]
```diff
@@ -2011,3 +2031,17 @@ func applyMessage(
...
+
+	return slotKey + ":" + proposerIndexKey
+}
+
+// Turns a uint64 value to a string representation.
+func uintToString(val uint64) string {
```
Comment (zzzckck) [inline consensus/parlia/parlia.go:2045]: seems no need

File: consensus/parlia/parlia.go [hunk 3]
```diff
@@ -2011,3 +2030,12 @@ func applyMessage(
 	}
 	return msg.Gas() - returnGas, err
 }
+
+// proposalKey build a key which is a combination of the block number and the proposer address.
+func proposalKey(header types.Header) string {
```
Comment (zzzckck) [inline consensus/parlia/parlia.go:2035]: double sign is not for sign at same block height, but sign on same block, e.g. with same ParentHash.
it should be: 
```
return header.ParentHash.String()+header.Coinbase.String()
```

File: consensus/parlia/parlia.go [hunk 4]
```diff
@@ -263,6 +267,10 @@ func New(
 	if err != nil {
 		panic(err)
 	}
+	recentHeaders, err := lru.NewARC(inMemorySignatures)
```
Comment (zzzckck) [inline consensus/parlia/parlia.go:270]: inMemorySignatures = 256, is too small, double sign would trace back to recent 86400 blocks.
better to set the cache size to 86400

**Review by zzzckck — APPROVED** [review]

**Review by galaio — APPROVED** [review]

File: consensus/parlia/parlia.go [hunk 5]
```diff
@@ -813,6 +823,16 @@ func (p *Parlia) verifySeal(chain consensus.ChainHeaderReader, header *types.Hea
 		return errUnauthorizedValidator(signer.String())
 	}
 
+	// check for double sign & add to cache
+	key := proposalKey(*header)
+
+	value, ok := p.recentHeaders.Get(key)
+	if ok {
+		doubleSignCounter.Inc(1)
```
Comment (zzzckck) [inline consensus/parlia/parlia.go:831]: pls add a warning message here

**Pull request merged**